aiohttp/client.py

```diff
@@ -383,7 +384,12 @@ def update_body_from_data(self, data):
             assert not isinstance(data, io.StringIO), \
                 'attempt to send text data instead of binary'
             self.body = data
-            self.chunked = True
+            if not self.chunked and isinstance(data, io.BufferedReader):
+                # Not chunking if content-length can be determined
+                self.headers[hdrs.CONTENT_LENGTH] = str(getsize(data.name))
```

Comment:
What if file was read a bit before it's going to  be sent with a request?

```
with open('foo.bar', 'rb') as f:
  magic = f.read(3)
  ...
  yield from aiohttp.request('post', 'http://localhost/upload', data=f)
```

In this case Content-Length header will declare more bytes than actually could be sent. Use tell() to figure out how much were already read. Also, since you have file description here, you can do just `os.fstat(data.fileno()).st_size` what is a bit more efficient.